src/c++/perf_analyzer/genai-perf/tests/test_wrapper.py

```diff
@@ -90,3 +93,31 @@ def test_service_openai(self, monkeypatch, arg):
 
         # Ensure the correct arguments are appended.
         assert cmd_string.count(" -i http") == 1
+
+    @patch("genai_perf.wrapper.subprocess.run")
+    def test_stdout_verbose(self, mock_subprocess_run):
+        args = MagicMock()
```

Comment:
what functionality does magicmock provide here?
sorry i am not super familiar

Reply:
It mocks the args. That way I don't have to use parse_args() or set up something implementation-specific with ArgParse(). So we can keep this test implementation-agnostic and only tests the one thing we want: that Profiler.run calls subprocess.run with the correct stdout value.

Reply:
awesome. thank you for explaining

Reply:
More specifically, we need the the args object for Profiler.run() in the next line. So this is creating a quick mock of them.